iceoryx2-bb/system-types/tests/base64url_tests.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use iceoryx2_bb_system_types::base64url::*;
+use iceoryx2_bb_testing::assert_that;
+
+#[test]
+fn base64url_new_with_legal_content_works() {
+    let sut = Base64Url::new(b"abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-_");
```

Comment:
For the sake of your future self, what happens if someone asks for a DOS platform and `FILENAME_LENGTH` is suddenly much shorter ... btw, when can I use iceoryx2 with VI on DOS?

Reply:
Whenever the time comes, and we have to support the 16-bit platform DOS we are screwed. Does it even has shared memory? It does not even have multiple processes, so why would you need IPC there?